ddtrace/internal/sampling.py

```diff
@@ -208,44 +211,90 @@ def apply_span_sampling_tags(self, span):
+        sampling_rules.append(sampling_rule)
+    return sampling_rules
+
+
+def get_span_sampling_json():
+    # type: () -> List[Dict[str, Any]]
+    env_json_rules = get_env_json()
+    file_json_rules = get_file_json()
+
+    if env_json_rules and file_json_rules:
+        log.warning(
+            (
+                "DD_SPAN_SAMPLING_RULES and DD_SPAN_SAMPLING_RULES_FILE detected. "
+                "Defaulting to DD_SPAN_SAMPLING_RULES value."
+            )
+        )
+        return env_json_rules
+    elif env_json_rules:
+        return env_json_rules
+    elif file_json_rules:
+        return file_json_rules
+    # No rules specified
+    else:
         return []
+
+
+def get_file_json():
+    # type: () -> Optional[List[Dict[str, Any]]]
+    file_json_raw = os.getenv("DD_SPAN_SAMPLING_RULES_FILE")
+    if file_json_raw:
+        with open(file_json_raw) as f:
+            return load_span_sampling_json(f.read())
     else:
-        sampling_rules = []
-        try:
-            json_rules = json.loads(json_rules_raw)  # type: List[SpanSamplingRules]
-            if not isinstance(json_rules, list):
-                raise TypeError("DD_SPAN_SAMPLING_RULES is not list, got %r" % json_rules)
-        except JSONDecodeError:
-            raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES=%r" % json_rules_raw)
-        for rule in json_rules:
-            if not isinstance(rule, dict):
-                raise TypeError("rule specified via DD_SPAN_SAMPLING_RULES is not a dictionary:%r" % rule)
-            # If sample_rate not specified default to 100%
-            sample_rate = float(rule.get("sample_rate", 1.0))
-            service = rule.get("service")
-            name = rule.get("name")
-            # If max_per_second not specified default to no limit
-            max_per_second = int(rule.get("max_per_second", -1))
-            if service is None and name is None:
-                raise ValueError(
-                    "Neither service or name specified for single span sampling rule:%r,"
-                    "at least one of these must be specified" % rule
-                )
-            if service:
-                _check_unsupported_pattern(service)
-            if name:
-                _check_unsupported_pattern(name)
-
-            try:
-                sampling_rule = SpanSamplingRule(
-                    sample_rate=sample_rate, service=service, name=name, max_per_second=max_per_second
-                )
-            except Exception as e:
-                raise ValueError("Error creating single span sampling rule {}: {}".format(json.dumps(rule), e))
-            sampling_rules.append(sampling_rule)
-        return sampling_rules
+        return None
+
+
+def get_env_json():
+    # type: () -> Optional[List[Dict[str, Any]]]
+    env_json_raw = os.getenv("DD_SPAN_SAMPLING_RULES")
+    if env_json_raw:
+        return load_span_sampling_json(env_json_raw)
+    else:
+        return None
```

Comment:
```suggestion
    return None
```

Reply:
Ah it seemed like I was getting mypy errors from this before, so I added the else and they went away... but I guess it must've been something else, because it works now. Thanks!